strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/nuget/filter/NugetFilterODataParserTestCase.java

```diff
@@ -76,7 +77,104 @@ public void testSearchConjunction(@NugetRepository(repositoryId = REPOSITORY_REL
         selector.select("count(*)");
 
         QueryTemplate<Long, ArtifactEntry> queryTemplate = new OQueryTemplate<>(entityManager);
+	
+        assertThat(queryTemplate.select(selector)).isEqualTo(1L);
+    }
+
+    /**
+     * Tests that All Chars that should work do
+     */
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    @Transactional
+    public void testSearchChars(@NugetRepository(repositoryId = REPOSITORY_RELEASES) Repository repository,
+                                @NugetTestArtifact(repositoryId = REPOSITORY_RELEASES,
+                                                   id = "Org.Carlspring.Strongbox.Nuget.Test.Nfpt_-test",
+                                                   versions = { "1.0.0",
+                                                                "1.0.1",
+                                                                "1.0.2" })
+                                Path artifactPath)
+    {
+        Selector<ArtifactEntry> selector = new Selector<>(ArtifactEntry.class);
+
+        NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(
+                "tolower(Id) eq 'org.carlspring.strongbox.nuget.test.nfpt_-test' and IsLatestVersion");
+        Predicate predicate = t.parseQuery().getPredicate();
+
+        selector.where(predicate)
+                .and(Predicate.of(ExpOperator.EQ.of("storageId", repository.getStorage().getId())))
+                .and(Predicate.of(ExpOperator.EQ.of("repositoryId", repository.getId())));
+
+        selector.select("count(*)");
+
+        QueryTemplate<Long, ArtifactEntry> queryTemplate = new OQueryTemplate<>(entityManager);
+	
         assertThat(queryTemplate.select(selector)).isEqualTo(1L);
     }
 
+    /*
+     * Tests that starting dash doesn't work
+     */
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    @Transactional
+    public void testSearchCharsStart(@NugetRepository(repositoryId = REPOSITORY_RELEASES) Repository repository,
+                                     @NugetTestArtifact(repositoryId = REPOSITORY_RELEASES,
+                                                        id = "-Org.Carlspring.Strongbox.Nuget.Test.Nfpt",
+                                                        versions = { "1.0.0",
+                                                                     "1.0.1",
+                                                                     "1.0.2" })
+                                     Path artifactPath)
+    {
+        Selector<ArtifactEntry> selector = new Selector<>(ArtifactEntry.class);
+
+        NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(
+                "tolower(Id) eq '-org.carlspring.strongbox.nuget.test.nfpt'");
```

Comment:
it's seems this query parsing should fail because of `VALUE` can't starts with `-`. 

@grantfar am I missing something why does it successfully parsed?